### ISSUE TITLE
Need first-class man .xhtml generation code

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2940]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)| @radix reported|
|-|-|
|Trac ID|trac#2940|
|Type|enhancement|
|Created|2007-12-09 17:39:46Z|
|Branch|https://github.com/twisted/twisted/tree/man-builder-2940|

admin/process-docs generates .xhtml and then .html files from our man pages. We need to do this in a way that's well -tested and -documented with python code. Then we should distribute the output both on the web site and with the release tarball.

Unfortunately this requires fixing a bug in lore in its man page processor (ticket to be linked here soon).

<details><summary>Searchable metadata</summary>

```
trac-id__2940 2940
type__enhancement enhancement
reporter__radix radix
priority__highest highest
milestone__twisted_8_0 twisted-8.0
branch__branches_man_builder_2940 branches/man-builder-2940
branch_author__ 
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__ 
time__1197221986000000 1197221986000000
changetime__1203327077000000 1203327077000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

What's the problem with lore exactly? Is it related to [#10856](https://github.com/twisted/twisted/issues/10856)?

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_2"></a>|@radix commented|
|-|-|

This ticket isn't actually blocked on that other issue. It can be done independently. 

I think therve is investigating the other issue now which will hopefully result in another ticket.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

(In [22423]) Branching to 'man-builder-2940'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve commented|
|-|-|

(In [22424]) Basic implementation of ManBuilder.

Refs [#2940](https://github.com/twisted/twisted/issues/2940)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve commented|
|-|-|

The lore issue is [#3017](https://github.com/twisted/twisted/issues/3017).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set owner to @radix|
|-|-|

Alright, this deserves a first look to check if it's in the right direction.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_8"></a>|@radix set owner to @therve|
|-|-|

1.I don't think we need a deleteInput option. We don't delete man files after converting them to lore.
 1. The docstring of ManBuilder.build could be improved. The summary should probably be "Generate Lore input files from man files."

Looks good! Thanks for working on this.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve commented|
|-|-|

(In [22580]) Process review comments.

Refs [#2940](https://github.com/twisted/twisted/issues/2940)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_10"></a>|@therve set owner to @radix|
|-|-|

Done!

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_11"></a>|@radix set owner to @therve|
|-|-|

Please fix the >80 column lines in test_release.py. After that you can merge. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_12"></a>|@therve commented|
|-|-|

(In [22583]) Fit into 80 columns.

Refs [#2940](https://github.com/twisted/twisted/issues/2940)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_13"></a>|@therve set status to `closed`|
|-|-|

(In [22584]) Merge man-builder-2940

Author: therve
Reviewer: radix
Fixes [#2940](https://github.com/twisted/twisted/issues/2940)

Add a tested object to generate Lore files from man pages.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_14"></a>|@therve commented|
|-|-|

(In [22585]) Revert r22584: windows slaves failures.

Refs [#2940](https://github.com/twisted/twisted/issues/2940)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_15"></a>|@therve set status to `reopened`|
|-|-|

```
[FAIL]: twisted.python.test.test_release.ManBuilderTestCase.test_build

Traceback (most recent call last):
  File "c:\twistedbuildbot24\WXP32-full2.5-scmikes-select\Twisted\twisted\python\test\test_release.py", line 535, in test_build
    self.assertEquals(output, "<html><head>\n<title>MANHOLE.1</title>"
twisted.trial.unittest.FailTest: not equal:
a = '<html><head>\r\n<title>MANHOLE.1</title></head>\r\n<body>\r\n\r\n<h1>MANHOLE.1</h1>\r\n\r\n<h2>NAME</h2>\r\n\r\n<p>manhole - Connect to a Twisted Manhole service\r\n</p>\r\n\r\n<h2>SYNOPSIS</h2>\r\n\r\n<p><strong>manhole</strong> </p>\r\n\r\n<h2>DESCRIPTION</h2>\r\n\r\n<p>manhole is a GTK interface to Twisted Manhole services. You can execute python\r\ncode as if at an interactive Python console inside a running Twisted process\r\nwith this.</p>\r\n\r\n</body>\r\n</html>\r\n'
b = '<html><head>\n<title>MANHOLE.1</title></head>\n<body>\n\n<h1>MANHOLE.1</h1>\n\n<h2>NAME</h2>\n\n<p>manhole - Connect to a Twisted Manhole service\n</p>\n\n<h2>SYNOPSIS</h2>\n\n<p><strong>manhole</strong> </p>\n\n<h2>DESCRIPTION</h2>\n\n<p>manhole is a GTK interface to Twisted Manhole services. You can execute python\ncode as if at an interactive Python console inside a running Twisted process\nwith this.</p>\n\n</body>\n</html>\n'
```

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_16"></a>|@therve set owner to @radix<br>@therve set status to `new`|
|-|-|

Just in case of.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_17"></a>|@radix set owner to @therve|
|-|-|

+1

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_18"></a>|@therve set status to `closed`|
|-|-|

(In [22591]) Merge man-builder-2940

Author: therve
Reviewer: radix
Fixes [#2940](https://github.com/twisted/twisted/issues/2940)

Add a tested object to generate Lore files from man pages.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
